### ISSUE TITLE
Achievements

### ISSUE THREAD

**Issue by agrande** [agrande/lba2remake#134]
- Discreet achievement system (should not break immersion)
- Encourages exploring the whole game
- Sharing across social media
- Achievements based on progress (hits by fighting, magicball, darts, penguin, (...), deaths, falls)
- User-created achievements

**Comment by felipesanches** [comment]
yeah! Something like this, but with other goals:

![achievements clash royale](https://user-images.githubusercontent.com/213676/36878259-d775c7f6-1d9c-11e8-9214-9eab8009bcc1.jpeg)

Would be cool to even come up with a list here (for prototyping) of which would be cool achievements on the LBA2 game.

**Comment by felipesanches** [comment]
I like the concept in general (simple goals to achieve). And I also like the specific implementation style adopted in games like Clash of Clans and Cut The Rope (not only goals, but with a system of 1, 2 or 3 stars depending on how good was the "quality" of your achievement).

In terms of code, I can think of 2 approaches:
* (a) each achievement consisting of a custom method that evaluates whether or not the goal was achieved. It would have to be running from time to time to update the user score.
* (b) an achievements registry where each goal is tied to an achievement_id and/or respective callback and then the game code itself would emit calls to the achievement system whenever a certain precondition was met.

**Comment by agrande** [comment]
I like the 1, 2, 3 stars idea, but not totally sure how relevant it would be to LBA.
In case of acheivements related to action sequences, like how fast you beat up some enemies, that would make sense, but not when just completing some parts related to the quests.
That is unless we consider how much time you spent completing a quest. But I don't think that would be a good idea to encourage players to rush through the game, unless they are speedrunners (see #132).

I guess we could have a mix of both types of achievements.

I'm not sure about the code approach. (a) seems cleaner at first sight, since it's not interfering with the game logic, but it could become more complex as some evaluations may require jumping through some hoops to implement.

**Comment by Jeffroiscool** [comment]
Not sure what achievements we want yet but humping the cow def should be one :D

**Comment by felipesanches** [comment]
Lets make a list ? It would at least be fun to write such a list with examples of possible achievements to have.

**Comment by aurelienfalco** [comment]
Find the secret room filled with Meca Penguins in the statue of Darkmonk! :)

**Comment by agrande** [comment]
Yep, that one sounds pretty obvious!

**Comment by felipesanches** [comment]
gave up on the idea?

**Comment by agrande** [comment]
No, it's just that we moved our TODO list to another system instead of using github issues for that.
We now use github issues exclusively for user feedback. I closed this issue as I was the one opening it, but I suppose I should keep it open as it was commented a lot. Although this is a rather low priority task, I must say...

**Comment by felipesanches** [comment]
ok, cool! Where can we see the new issue tracker so that we can help as well?

**Comment by agrande** [comment]
Well, we have decided to keep it private between frequent contributors, so we can keep it focused and efficient, but you can still open issues here if you have some feature request or want to discuss something in particular. And if you want to help with code we can discuss things in the Slack channel.

**Comment by xesf** [comment]
LBA1 Enhanced Edition Achievements list:
https://steamdb.info/app/397330/stats/
